### PR TITLE
 Blog app - Views

### PR DESCRIPTION
**Pull request by Marcraphael12**
### Project requirements
- Implement the design from the sneak peek wireframes.
    - Styling is nice to have but focus on making sure everything functions first. (Note: Ignore login logic, you would be using Devise in coming milestone)

**Note: Analyze the wireframes deeply and try to identify all the necessary elements and how a user would navigate from one page to another**

### CODE REVIEW

File: app/views/posts/index.html.erb [hunk 1]
```diff
@@ -1,17 +1,37 @@
...
+            <div class="commets-container">
+              <h4>Comments:</h4>
+              <ul>
+                <% post.recent_comments.each do |comment| %>
+                  <p><%= comment.author.Name %>: <%= comment.Text %></p>
+                <% end %>
+              </ul>
+            </div>
+          </li>
+        <% end %>
```
Comment (bienvenuushindi) [inline app/views/posts/index.html.erb:33]: :red_circle: [REQUIRED] The Ruby naming convention for methods and variables is snake_case. So, the following names should be changed:

- `CommentsCounter` to `comments_counter`
- `LikesCounter` to `likes_counter`
- `Name` to `name`
- `Text` to `text`
- `Title` to `title`

Note: Kindly check this[ list of ruby best practices](https://github.com/microverseinc/curriculum-ruby/blob/main/articles/ruby_best_practices.md)

File: app/views/posts/show.html.erb [hunk 2]
```diff
@@ -1,3 +1,17 @@
...
+      <p><%= @post.Text %></p>
+      <span>Likes: <%= @post.LikesCounter %> </span>
+    </div>
+    <div class="commets-container">
+      <h4>Comments:</h4>
+      <ul>
+        <% @post.recent_comments.each do |comment| %>
+          <p><%= comment.Text %></p>
+        <% end %>
+      </ul>
```
Comment (bienvenuushindi) [inline app/views/posts/show.html.erb:14]: :red_circle: [REQUIRED] The Ruby naming convention for methods and variables is snake_case. So, the following names should be changed:

- `LikesCounter` to `likes_counter`
- `Text` to `text`
- `Title` to `title`


Note: Kindly check this[ list of ruby best practices](https://github.com/microverseinc/curriculum-ruby/blob/main/articles/ruby_best_practices.md)

File: app/views/users/index.html.erb [hunk 3]
```diff
@@ -1,17 +1,17 @@
...
+            <div class="user-image border"></div>
+            <div class="user-data border">
+              <h1><%= user.Name %></h1>
+              <p class="number-posts">Number of posts: <%= user.PostCounter %></p>
+            </div>
+          </li>
+        <% end %>
+      <% end %>
+    </ul>  
+  </section>
```
Comment (bienvenuushindi) [inline app/views/users/index.html.erb:16]: :red_circle: [REQUIRED] The Ruby naming convention for methods and variables is snake_case. So, the following names should be changed:

- `PostsCounter` to `posts_counter`
- `Name` to `name`

Note: Kindly check this[ list of ruby best practices](https://github.com/microverseinc/curriculum-ruby/blob/main/articles/ruby_best_practices.md)

File: app/views/users/show.html.erb [hunk 4]
```diff
@@ -1,3 +1,34 @@
...
+            <div class="right">
+              <span>Comments: <%= post.CommentsCounter %></span>
+              <span class="left">Likes: <%= post.LikesCounter %> </span>
+            </div>
+          </li>
+        <% end %>
+      <% end %>
+    </ul>
+    <%= link_to "See all posts", user_posts_path(user_id: @user.id), class: "btn" %>
+  </section>
```
Comment (bienvenuushindi) [inline app/views/users/show.html.erb:33]: :red_circle: [REQUIRED] The Ruby naming convention for methods and variables is snake_case. So, the following names should be changed:

- `CommentsCounter` to `comments_counter`
- `LikesCounter` to `likes_counter`
- - `PostCounter` to `post_counter`
- `Name` to `name`
- `Text` to `text`
- `Title` to `title`
- `Bio` to `bio`

Note: Kindly check this[ list of ruby best practices](https://github.com/microverseinc/curriculum-ruby/blob/main/articles/ruby_best_practices.md)

File: app/views/users/index.html.erb [hunk 5]
```diff
@@ -1,17 +1,17 @@
...
-    </li>
-  </ul>
-</div>
+<main>
+  <section>
+    <ul>
+      <% @users.each do |user|%>
+        <%= link_to user do %>
+          <li class="card hover">
+            <div class="user-image border"></div>
```
Comment (bienvenuushindi) [inline app/views/users/index.html.erb:7]: :red_circle: [REQUIRED] Please make sure you show the user's image in the div element. The image_tag method can be used to render an HTML image tag.

Here is the corrected code:

```
<div class="user-image border">
  <%= image_tag(user.photo, class: "add-css-class") %>
</div>
```

File: app/views/posts/index.html.erb [hunk 6]
```diff
@@ -1,17 +1,37 @@
...
-    </li>
-    <li>
-      we are all posts hey 😒
-    </li>
-  </ul>
-</div>
+<main>
+  <section>
+    <div class="card">
+      <div class="user-image border"></div>
```
Comment (bienvenuushindi) [inline app/views/posts/index.html.erb:4]: :red_circle: [REQUIRED] Please make sure you show the user's image in the div element. The image_tag method can be used to render an HTML image tag.

Here is the corrected code:

```
<div class="user-image border">
  <%= image_tag(user.photo, class: "add-css-class") %>
</div>
```

File: app/views/users/index.html.erb [hunk 7]
```diff
@@ -1,17 +1,17 @@
...
-    </li>
-    <li>
-      Rita Nkem Danniel
-    </li>
-  </ul>
-</div>
+<main>
+  <section>
+    <ul>
+      <% @users.each do |user|%>
```
Comment (bienvenuushindi) [inline app/views/users/index.html.erb:4]: :orange_circle: [OPTIONAL] :bulb: Did you know :thinking: that this code can be improved  `@user.each do ...end`?

The good practice is to use the render method. This method takes a collection of objects as an argument and renders a partial template for each object in the collection. The partial template is a separate file that contains the HTML code for displaying a single object.

Here is how you would implement the good practice in your code:
Replace the bloc 
```
 @user.each do
      # your code
  end
```

 with this  single line 
```
# app/views/users/index.html.erb

<%= render partial: "user", collection: @users %>
```

Then create a partial file` app/views/users/_user.html.erb`  and move the content inside the  `@user.each do ...end` bloc in the new file

```
# app/views/users/_user.html.erb

     #   HTML Code Here displaying
     #   user.photo, user.name and user.posts_counter

 <%= link_to user do %>
     <li class="card hover">
         <div class="user-image border"></div>
        <div class="user-data border">
              <h1><%= user.Name %></h1>
              <p class="number-posts">Number of posts: <%= user.PostCounter %></p>
       </div>
    </li>
 <% end %>
```

**Review by bienvenuushindi — CHANGES_REQUESTED** [review]
## Hi @Marcraphael12 ,

Good job so far!

## Highlights!
- GitFlow ✔️ 
- No linter errors ✔️ 
- Descriptive pull request ✔️ 
- Good code organization and working app ✔️
- Descriptive README ✔️ 

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
_Check the comments under the review and here 👇._


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@julie-ify` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/users/index.html.erb (hunk 7 above)
Comment (Marcraphael12) [reply app/views/users/index.html.erb:4]: Thanks for this amaizing suggestion... i'll be happy to apply in my free time

**Review by descholar-ceo — APPROVED** [review]
Hi @Marcraphael12 🤝 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
👍🏽 Implemented the changes requested from the first review ✔️ 
👍🏽 No new issues introduced while implementing the comments ✔️ 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**